includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1080,17 +1100,20 @@ private function check_matching_attribute_value( $attr_name, $attr_value, $spec_
 	 *                                        is no rule for this attribute.
 	 */
 	private function check_attr_spec_rule_value_casei( $node, $attr_name, $attr_spec_rule ) {
-		/**
-		 * Check 'value_casei' - case insensitive
-		 */
-		if ( isset( $attr_spec_rule[ AMP_Rule_Spec::VALUE_CASEI ] ) ) {
-			$rule_value = strtolower( $attr_spec_rule[ AMP_Rule_Spec::VALUE_CASEI ] );
+		if ( ! isset( $attr_spec_rule[ AMP_Rule_Spec::VALUE_CASEI ] ) ) {
+			return AMP_Rule_Spec::NOT_APPLICABLE;
+		}
+
+		$result      = AMP_Rule_Spec::NOT_APPLICABLE;
+		$rule_values = (array) $attr_spec_rule[ AMP_Rule_Spec::VALUE_CASEI ];
+		foreach ( $rule_values as $rule_value ) {
+			$rule_value = strtolower( $rule_value );
 			if ( $node->hasAttribute( $attr_name ) ) {
 				$attr_value = strtolower( $node->getAttribute( $attr_name ) );
 				if ( $attr_value === (string) $rule_value ) {
 					return AMP_Rule_Spec::PASS;
 				} else {
-					return AMP_Rule_Spec::FAIL;
+					$result = AMP_Rule_Spec::FAIL;
```

Comment:
Good question. The reason is that the spec changed from `value_casei` being a scalar value to being an array. So instead of testing a single value, we now need to check if _any_ match. You can see in the `if` condition immediately preceding that it will `return` immediately upon a match. But then if it does not match then this `FAIL` will end up being the return value after all of the values are matched. But if the element doesn't have the attribute at all, then the `NOT_APPLICABLE` value is returned.

Reply:
@westonruter Thank you for clarifying.  That makes sense.